packages/core/lib/queues/AbstractQueueService.ts

```diff
@@ -42,11 +43,15 @@ export abstract class AbstractQueueService<
   ExecutionContext = undefined,
   PrehandlerOutput = undefined,
 > {
+  // Used to keep track of the number of retries performed on consumer
+  private readonly messageNumberOfRetriesField = '_internalNumberOfRetries'
```

Comment:
`retry` naming sounds confusing for me in this context, because it's not clear how we are counting them - so first attempt is retry zero, and then second attempt is retry 1?

BullMQ naming of `attempts` is clearer for me; we can start at 1 (default when field is not set) and apply consistent counting logic.

Reply:
Sounds good to me! will apply the change thanks Igor 🙇

Reply:
Mmm, thinking about it, I am not really sure, this counting will only be applied on retryLater cases, meaning that errors will go with the SQS usual retry logic flow. 

`retry` here means, the number of times the consumer finished with `retryLater` result, naming it `attempt` I think can be more confusing, maybe something `_internalNumberOfRetryLater` to make it more clear?

Reply:
fair point. maybe we can explain this with a jsdoc, without naming changes?

Reply:
Of course! I will add it in a bit :D thanks Igor 🙏